connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -702,6 +706,9 @@ private void startConnector(String connectorName) {
         ConnectorContext ctx = new HerderConnectorContext(DistributedHerder.this, connName);
         worker.startConnector(connConfig, ctx, this);
 
+        if (tempConnectors.containsKey(connName)) {
```

Comment:
The intention is to remove the temporary connector instance if we submit a connector with the same name for execution.